### PR TITLE
[RFC/RDY] flent: add manpage + init http-getter

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change
without at least netperf the program is useless since it is an interface to netperf.
http-getter is a small utility used just by flent so I addded it to the flent default.nix.

NB: for those testing it, I kept the stable release but there is a nasty bug that was solved there
https://github.com/tohojo/flent/issues/137 so you might need to override the src to sthg more recent.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/flent/default.nix [hunk 1]
```diff
@@ -1,4 +1,33 @@
-{ stdenv, buildPythonApplication, fetchFromGitHub, matplotlib, netperf, procps, pyqt5 }:
+{ stdenv, buildPythonApplication, fetchFromGitHub, matplotlib, netperf, procps, pyqt5
+, makeWrapper, iperf2
+, withMan ? true, sphinx ? null
```
Comment (xeji) [inline pkgs/applications/networking/flent/default.nix]: I don't think it's worth making the manual optional. Doesn't take long to build (and hydra does it for you anyway), and it's put in a separate output. So I suggest to remove this option and always build the manual, which is the de-facto standard in nixpkgs.

File: pkgs/applications/networking/flent/default.nix [hunk 2]
```diff
@@ -10,13 +39,25 @@ buildPythonApplication rec {
...
+  buildInputs = [ makeWrapper ] ++ stdenv.lib.optionals withMan [ sphinx ];
+
   propagatedBuildInputs = [
     matplotlib
     procps
     pyqt5
   ];
 
+  postBuild = stdenv.lib.optionalString withMan ''
+    make man
```
Comment (xeji) [inline pkgs/applications/networking/flent/default.nix]: This `postBuild` is not necessary and can be removed. The manual is built automatically.

File: pkgs/applications/networking/flent/default.nix (hunk 1 above)
Comment (teto) [reply pkgs/applications/networking/flent/default.nix]: I received different feedback on other PRs but I definitely agree we should encourage manpages so I build it by default.

**Comment by teto** [comment]
In the end I removed the wrapping since it might pull too much otherwise and wouldn't work in python development mode which was my interest.  The PR juste enables the manpage and add http-getter to cover some tests in flent.

So people should keep using it describing the necessary dependencies `nix-shell -p flent iperf http-getter` etc

**Comment by mmlb** [comment]
>In the end I removed the wrapping since it might pull too much otherwise and wouldn't work in python development mode which was my interest.

hey @teto what ddi you mean by python-development mode, do you recall? Maybe we should do 2 packages here, flent and flent-full or something? I'd like to have all the needed tools depended on and brought in somehow.

**Comment by teto** [comment]
@mmlb I meant I was using flent in a nix-shell (equivalent to `python setup.py develop`).  So the executable is not wrapped (maybe there is some trick to wrap it).
I think we should add an extraLibs settings like in i3pystatus
https://github.com/NixOS/nixpkgs/blob/47b551c6a854a049045455f8ab1b8750e7b00625/pkgs/applications/window-managers/i3/pystatus.nix#L1 , default it to a full setting that people can override.

**Pull request merged**